source/light/contracts/Light.sol

```diff
@@ -93,51 +104,65 @@ contract Light is ILight {
    * @param signerAmount Amount for signerToken
    * @param senderToken Contract address of the ERC20 token that will be transferred from the sender
    * @param senderAmount Amount for senderToken
-   * @param signature Signature generated by the signer
    */
   function swap(
     uint256 nonce,
     uint256 expiry,
+    address signerWallet,
     IERC20 signerToken,
     uint256 signerAmount,
     IERC20 senderToken,
     uint256 senderAmount,
-    bytes calldata signature
+    uint8 v,
```

Comment:
Optional: you can stash this uint8 in the high bits of an `address` to save 32 bytes of calldata.

Reply:
thought about it but most of that calldata will be 0s and would make it a little bit harder to read